tsz-compress/src/v2/tests.rs

```diff
@@ -0,0 +1,902 @@
+        let values = vec![-3, 2, 0, 1, 2, -3, -1, -2, -4, -3];
+        let (queue_size, num_emitted_samples, decoded_size, decoded_values) =
+            _can_encode_decode_delta_values_i16(&values, false);
+        assert_eq!(num_emitted_samples, 10);
+        assert_eq!(queue_size, values.len() - num_emitted_samples);
+        assert_eq!(
+            values[..num_emitted_samples],
+            decoded_values[..decoded_size]
+        );
+    }
+
+    #[test]
+    fn can_encode_decode_delta_i16_sanity2() {
+        // Case 4 and 5: Encode and decode 10 samples between [-8, 7] in 3 bits
+        let values = vec![-4, 6, -8, 3, 2, -1, 7, 0, -5, 4];
+        let (queue_size, num_emitted_samples, decoded_size, decoded_values) =
+            _can_encode_decode_delta_values_i16(&values, false);
+        assert_eq!(num_emitted_samples, 8);
+        assert_eq!(queue_size, values.len() - num_emitted_samples);
+        assert_eq!(
+            values[..num_emitted_samples],
+            decoded_values[..decoded_size]
+        );
+    }
+
+    #[test]
+    fn can_encode_decode_delta_i16_sanity3() {
+        // Case 3, 4, 5
+        let values = vec![-32, 115, -78, 56, 12, -127, 89, 43, -3, 101];
+        let (queue_size, num_emitted_samples, decoded_size, decoded_values) =
+            _can_encode_decode_delta_values_i16(&values, false);
+        assert_eq!(num_emitted_samples, 4);
+        assert_eq!(queue_size, values.len() - num_emitted_samples);
+        assert_eq!(
+            values[..num_emitted_samples],
+            decoded_values[..decoded_size]
+        );
+    }
+
+    #[test]
+    fn can_encode_decode_delta_i16_sanity4() {
+        // Case 2, 3, 4, 5
+        let values = vec![-256, 489, -123, 402, 67, -505, 311, 109, -412, 210];
+        let (queue_size, num_emitted_samples, decoded_size, decoded_values) =
+            _can_encode_decode_delta_values_i16(&values, false);
+        assert_eq!(num_emitted_samples, 3);
+        assert_eq!(queue_size, values.len() - num_emitted_samples);
+        assert_eq!(
+            values[..num_emitted_samples],
+            decoded_values[..decoded_size]
+        );
+    }
+
+    #[test]
+    fn can_encode_decode_delta_i16_sanity5() {
+        // Case 1, 2, 3, 4, 5
+        let values = vec![
+            -32768, 23456, -7891, 16042, 5678, -27600, 9123, 14567, -22222, 7890,
+        ];
+        let (queue_size, num_emitted_samples, decoded_size, decoded_values) =
+            _can_encode_decode_delta_values_i16(&values, false);
+        assert_eq!(num_emitted_samples, 2);
+        assert_eq!(queue_size, values.len() - num_emitted_samples);
+        assert_eq!(
+            values[..num_emitted_samples],
+            decoded_values[..decoded_size]
+        );
+    }
+
+    #[test]
+    fn can_encode_decode_delta_random_i16() {
+        // Random values in i16 range
+        let mut rng = rand::thread_rng();
+        for _ in 0..=100000 {
```

Comment:
encode is working on a 10 element buffer, but decode needs to be tested over one shot decoding the whole column.